File: back/exampleGenerator.py
from random import randint

def getSelectImgOfDice(nb):
    return f"/dices/dice{nb}_select.png"

def getImgOfDice(nb):
    return f"/dices/dice{nb}.jpg"

def generateComplexeExample(combi):
    combi["threeOfAKind"] = generateThreeOfAKind()
    combi["fourOfAKind"] = generateFourOfAKind()
    combi["fullHouse"] = generateFull()
    combi["smallStraight"] = generateSmallStraight()
    combi["largeStraight"] = generateLargeStraight()
    combi["yahtzee"] = generateYahtzee()
    combi["chance"] = generateChance()
    return combi

def generateThreeOfAKind():
    triplicateNumb = randint(1,6)
    res = [getSelectImgOfDice(triplicateNumb) for i in range(3)]
    lastNumb = triplicateNumb
    for i in range(2):
        numb = randint(1,6)
        while numb == lastNumb:
            numb = randint(1,6)
        res.append(f"/dices/dice{numb}.jpg")
        lastNumb = numb
    return res

def generateFourOfAKind():
    quadrupleNumb = randint(1,6)
    res = [getSelectImgOfDice(quadrupleNumb) for _ in range(4)]
    
    numb = randint(1,6)
    while numb == quadrupleNumb:
        numb = randint(1,6)
    res.append(f"/dices/dice{numb}.jpg")
    return res

def generateFull():
    triplicateNumb = randint(1,6)
    res = [getImgOfDice(triplicateNumb) for _ in range(3)]

    numb = randint(1,6)
    while numb == triplicateNumb:
        numb = randint(1,6)
    res.append(getImgOfDice(numb))
    res.append(getImgOfDice(numb))

    return res

def generateSmallStraight():
    numb = randint(1,3)
    res = [getSelectImgOfDice(numb), getSelectImgOfDice(numb+1), getSelectImgOfDice(numb+2), getSelectImgOfDice(numb+3), getImgOfDice(numb)]
    return res

def generateLargeStraight():
    numb = randint(1,2)
    res = [getSelectImgOfDice(numb), getSelectImgOfDice(numb+1), getSelectImgOfDice(numb+2), getSelectImgOfDice(numb+3), getSelectImgOfDice(numb+4)]
    return res

def generateYahtzee():
    numb = randint(1,6)
    res = [getImgOfDice(numb) for _ in range(5)]
    return res

def generateChance():
    res = [getImgOfDice(randint(1,6)) for _ in range(5)]
    return res

File: src/logic/exampleGenerator.ts
function getSelectedImgOfDice(dice: number): string {
    return "/dices/dice" + dice + "_select.png";
}

function getImgOfDice(dice: number): string {
    return "/dices/dice" + dice + ".png";
}

export function generateComplexeExamples(combinations: Record<string, string[]>): Record<string, string[]> {
    combinations["threeOfAKind"] = generateKind(3);
    combinations["fourOfAKind"] = generateKind(4);
    combinations["fullHouse"] = generateFullHouse();
    combinations["smallStraight"] = generateSmallStraight();
    combinations["largeStraight"] = generateLargeStraight();
    combinations["yahtzee"] = generateYahtzee();
    combinations["chance"] = generateChance();
    return combinations;
}

function generateKind(n: number): string[] {
    let imgs: string[] = [];
    let numb = Math.floor(Math.random() * 6) + 1;
    for (let i = 0; i < n; i++) {
        imgs.push(getSelectedImgOfDice(numb));
    }

    for (let i = 0; i < 5 - n; i++) {
        let soloNumb = Math.floor(Math.random() * 6) + 1;
        while (soloNumb === numb) {
            soloNumb = Math.floor(Math.random() * 6) + 1;
        }
        imgs.push(getImgOfDice(soloNumb));
    }

    return imgs;
}

function generateFullHouse(): string[] {
    let imgs: string[] = [];
    let numb1 = Math.floor(Math.random() * 6) + 1;
    let numb2 = Math.floor(Math.random() * 6) + 1;
    while(numb2 === numb1){
        numb2 = Math.floor(Math.random() * 6) + 1;
    }

    for (let i = 0; i < 3; i++) {
        imgs.push(getSelectedImgOfDice(numb1));
    }

    for (let i = 0; i < 2; i++) {
        imgs.push(getSelectedImgOfDice(numb2));
    }

    return imgs;
}

function generateSmallStraight(): string[] {
    let imgs: string[] = [];
    let numb1 = Math.floor(Math.random() * 3) + 1;
    let numb2 = numb1 + 1;
    let numb3 = numb2 + 1;
    let numb4 = numb3 + 1;

    let numbs = [numb1, numb2, numb3, numb4];

    for (let i = 0; i < 4; i++) {
        imgs.push(getSelectedImgOfDice(numbs[i]));
    }

    imgs.push(getImgOfDice(numb2));

    return imgs;
}

function generateLargeStraight(): string[] {
    let imgs: string[] = [];
    
    let numb1 = Math.floor(Math.random() * 1) + 1;
    let numb2 = numb1 + 1;
    let numb3 = numb2 + 1;
    let numb4 = numb3 + 1;
    let numb5 = numb4 + 1;

    let numbs = [numb1, numb2, numb3, numb4, numb5];

    for (let i = 0; i < 5; i++) {
        imgs.push(getSelectedImgOfDice(numbs[i]));
    }

    return imgs;
}


function generateYahtzee(): string[] {
    let imgs: string[] = [];
    let numb = Math.floor(Math.random() * 6) + 1;
    for (let i = 0; i < 5; i++) {
        imgs.push(getSelectedImgOfDice(numb));
    }

    return imgs;
}

function generateChance(): string[] {
    let imgs: string[] = [];
    for (let i = 0; i < 5; i++) {
        let numb = Math.floor(Math.random() * 6) + 1;
        imgs.push(getImgOfDice(numb));
    }

    return imgs;
}
